controller/sleep_manager.py

```diff
@@ -0,0 +1,473 @@
+import asyncio
+import time
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Set
+
+import aiohttp
+from traffic_monitor import TrafficMonitor
+
+from kvcached.utils import get_kvcached_logger
+
+logger = get_kvcached_logger()
+
+
+@dataclass
+class SleepConfig:
+    """Configuration for sleep mode management"""
+    idle_threshold_seconds: int = 300  # 5 minutes
+    check_interval_seconds: int = 60  # Check every minute
+    auto_sleep_enabled: bool = False  # Whether to automatically put models to sleep
+    wakeup_on_request: bool = True  # Whether to automatically wake models on request
+    min_sleep_duration: int = 60  # Minimum time to keep model asleep (seconds)
+    vllm_models_config: Dict[str, Dict[
+        str,
+        str]] = None  # model_name -> {"host": "localhost", "port": "8000"}
+    sglang_models_config: Dict[str, Dict[
+        str,
+        str]] = None  # model_name -> {"host": "localhost", "port": "8000"}
+
+
+class SleepManager:
+    """Manages sleep mode for idle models to save resources"""
+
+    def __init__(self,
+                 config: Optional[SleepConfig] = None,
+                 traffic_monitor: Optional[TrafficMonitor] = None):
+        self.config = config or SleepConfig()
```

Comment:
Better to support parsing the .yaml file to get these configs. I mean when using launch.py to launch everything.